Tasks/JavaToolInstaller/task.json

```diff
@@ -0,0 +1,157 @@
+{
+    "id": "C0E0B74F-0931-47C7-AC27-7C5A19456A36",
+    "name": "JavaToolInstaller",
+    "friendlyName": "Java Tool Installer",
+    "description": "Acquires a specific version of Java from a user supplied Azure blob or the tools cache and sets JAVA_HOME. Use this task to change the version of Java used in Java tasks.",
+    "helpMarkDown": "[More Information]()",
+    "category": "Tool",
+    "runsOn": [
+        "Agent",
+        "DeploymentGroup"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 126,
+        "Patch": 0
+    },
+    "preview": true,
+    "satisfies": ["Java"],
+    "demands": [],
+    "groups": [
+        {
+            "name":"JavaInAzureGroup",
+            "displayName": "Download Java from an Azure blob",
+            "isExpanded": true
+        }
+    ],
+    "instanceNameFormat": "Use Java $(versionSpec)",
+    "inputs": [
+        {
+            "name": "versionSpec",
+            "type": "string",
+            "label": "JDK Version",
+            "defaultValue": "1.8",
+            "required": true,
+            "helpMarkDown": "A number that specifies the JDK version to make available on the path."
+        },
+        {
+            "name": "jdkArch",
+            "type": "pickList",
+            "label": "JDK Arch type",
+            "options": {
+                "x64": "X64",
+                "x86": "X86"
+            },
+            "required": true,
+            "helpMarkDown": "The bit version of the JDK."
+        },
+        {
+            "name": "jdkSource",
+            "type": "pickList",
+            "label": "JDK source",
+            "required": true,
+            "options": {
+                "AzureStorage": "Azure Storage",
+                "LocalDirectory": "Local Directory"
+            },
+            "helpMarkDown": "Source for the compressed JDK."
+        },
+        {
+            "name": "jdkPath",
+            "type": "filePath",
+            "label": "JDK path",
+            "required": true,
+            "visibleRule": "jdkSource == LocalDirectory",
+            "helpMarkDown": "Path to where the compressed JDK is located. The path could be in your source repository or a local path on the agent."
+        },
+        {
+            "name": "azureResourceManagerEndpoint",
+            "type": "connectedService:AzureRM",
+            "label": "Azure Subscription",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Choose the Azure Resource Manager subscription for the JDK."
+        },
+        {
+            "name": "azureStorageAccountName",
```

Comment:
`azureStorageAccountName` as you have here is probably fine.  Usually we add the suffix `Option` for pickLists, but in this case 'Name' is probably more correct.